src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeModal.test.ts

```diff
@@ -326,16 +329,17 @@ describe('GcpComputeModal', () => {
       };
       const runtime = getGoogleRuntime(runtimeProps);
 
-      const runtimeFunc = jest.fn(() => ({
-        create: jest.fn(),
-        details: () => runtime,
-      }));
-      asMockedFn(Ajax).mockReturnValue({
-        ...defaultAjaxImpl,
-        Runtimes: {
-          runtime: runtimeFunc as Partial<RuntimeAjaxContractV1>,
-        } as Partial<RuntimesAjaxContract>,
-      } as AjaxContract);
+      const runtimeFunc: RuntimesAjaxContract['runtime'] = jest.fn(() =>
+        partial<RuntimeAjaxContractV1>({
+          create: jest.fn(),
+          details: async () => runtime,
```

Comment:
the improved type safety I added exposed some signature mismatches, such as details function here needing to return a Promise.